src/yadbox/export.py

```diff
@@ -29,23 +29,58 @@ def dump_pineappl_to_file(output, filename, obsname):
     interpolation_polynomial_degree = output["polynomial_degree"]
     lepton_pid = output["projectilePID"]
 
-    # init pineappl objects
-    lumi_entries = [
-        pineappl.lumi.LumiEntry([(pid, lepton_pid, 1.0)]) for pid in output["pids"]
-    ]
+    # Instantiate the objecs required to construct a new Grid
+    channels = [pineappl.boc.Channel([([pid], 1.0)]) for pid in output["pids"]]
     first_esf_result = output[obsname][0]
-    orders = [pineappl.grid.Order(*o) for o in first_esf_result.orders]
+    orders = [pineappl.boc.Order(*o, 0) for o in first_esf_result.orders]
+    polarized = obsname.startswith("g")
+    convolution_types = pineappl.convolutions.ConvType(
+        polarized=polarized, time_like=False
+    )
+    convolutions = [
+        pineappl.convolutions.Conv(convolution_types=convolution_types, pid=2212)
+    ]
+    kinematics = [pineappl.boc.Kinematics.Scale(0), pineappl.boc.Kinematics.X(0)]
+    scale_funcs = pineappl.boc.Scales(
+        ren=pineappl.boc.ScaleFuncForm.Scale(0),
+        fac=pineappl.boc.ScaleFuncForm.Scale(0),
+        frg=pineappl.boc.ScaleFuncForm.NoScale(0),
+    )
     bins = len(output[obsname])
-    bin_limits = list(map(float, range(0, bins + 1)))
-    # subgrid params
-    params = pineappl.subgrid.SubgridParams()
-    params.set_reweight(False)
-    params.set_x_bins(len(interpolation_xgrid))
-    params.set_x_max(interpolation_xgrid[-1])
-    params.set_x_min(interpolation_xgrid[0])
-    params.set_x_order(interpolation_polynomial_degree)
+    bin_limits = pineappl.boc.BinsWithFillLimits.from_fill_limits(
+        fill_limits=list(map(float, range(0, bins + 1)))
+    )
+    interpolations = [
+        pineappl.interpolation.Interp(
+            min=1,
+            max=1e3,
```

Comment:
This refers to $Q^2$, right? (or more precisely $\mu_F^2$ - meaning the scale of the coefficient functions; here of course they coincide) Then we should read from the actual kinematics at hand, i.e. make it dynamic - as with $z$ below (=interpolation grid)

Reply:
done in 421e5cb78ee1907608f948189e037a5b95847211